File: 0004/euler0004.m
(*
    This should be self-explanatory.

    Output: 906609
*)


Print[
    Max @ Select[
        Flatten[Table[a * b, {a, 100, 999}, {b, a, 999}]],
        IntegerDigits[#] == Reverse[IntegerDigits[#]] &
    ]
];

File: 0004/euler0004.py
# This should be self-explanatory.

# Output: 906609

if __name__ == "__main__":
    max_prod = 0
    for a in range(100, 1000):
        for b in range(a, 1000):
            prod = a * b
            if str(prod) == str(prod)[::-1] and prod > max_prod:
                max_prod = prod
    print(max_prod)
